census/vanilla-tf/trainer/task.py

```diff
@@ -0,0 +1,338 @@
+  cluster_spec, job_name, task_index = parse_tf_config()
+
+  # Create and start a server
+  server = tf.train.Server(cluster_spec,
+                           job_name=job_name,
+                           task_index=task_index)
+
+  is_chief = (job_name == 'master')
+
+  if job_name == 'ps':
+    server.join()
+  elif job_name in ['master', 'worker']:
+    with tf.device(tf.train.replica_device_setter()):
+      inputs, labels = get_placeholders()
+      graph = make_graph(inputs, labels)
+
+    init = tf.global_variables_initializer()
+
+    with tf.train.MonitoredTrainingSession(master=server.target,
+                                           is_chief=is_chief,
+                                           checkpoint_dir=job_dir,
+                                           save_checkpoint_secs=20,
+                                           save_summaries_steps=50) as session:
+      session.run(init)
+      training_steps(session, graph, inputs, labels, max_steps,
+                     train_eval_tensor, job_name=job_name, job_id=task_index)
+
+
+def evaluate_accuracy(session, eval_step, inputs, labels,
+                      eval_input, eval_label):
+  """Perform the evaluation step to calculate accuracy."""
+  return 100 * session.run(
+      eval_step,
+      feed_dict={
+          inputs: session.run(eval_input),
+          labels: session.run(eval_label)
+      }
+  )
+
+
+def parse_tf_config():
+  """Parse TF_CONFIG to cluster_spec, job_name and task_index."""
+
+  tf_config = os.environ.get('TF_CONFIG')
+
+  if tf_config is None or tf_config == '':
+    return None
+
+  tf_config_json = json.loads(tf_config)
+
+  cluster = tf_config_json.get('cluster')
+  job_name = tf_config_json.get('task').get('type')
+  task_index = tf_config_json.get('task').get('index')
+
+  cluster_spec = tf.train.ClusterSpec(cluster)
+  return cluster_spec, job_name, task_index
+
+
+if __name__ == "__main__":
+  parser = argparse.ArgumentParser()
+  parser.add_argument(
+      '--train_data_path', required=True, type=str,
+      help='Training file location')
+  parser.add_argument(
+      '--eval_data_path', required=True, type=str,
+      help='Evaluation file location')
+  parser.add_argument(
+      '--job_dir', required=True, type=str,
+      help='Location to write checkpoints and export model'
+  )
+  parser.add_argument(
+      '--distributed', default=False, type=bool,
+      help='Run the code either as single or distributed mode')
+  parser.add_argument('--max_steps', type=int, default=200,
+      help='Maximum number of training steps to perform')
+  parse_args, unknown = parser.parse_known_args()
+
+  train_tensor = read_input_tensor(parse_args.train_data_path)
+
+  # Skip first row which has meta information
+  eval_tensor = read_input_tensor(parse_args.eval_data_path,
+                                  skiprows=[0])
+
+  train_eval_tensor = [train_tensor, eval_tensor]
+
+  # Start single node training
```

Comment:
No need for this flag. Lets make the code run in both modes, always, and the choice of single and distributed be an outcome of the cluster spec chosen when submitting the job.